rust/src/dataset/scanner.rs

```diff
@@ -84,6 +84,9 @@ pub struct Scanner {
     /// Scan the dataset with a meta column: "_rowid"
     with_row_id: bool,
 
+    /// Whether to scan in deterministic order (default: true)
+    ordered: bool,
```

Comment:
should we use `batch_ordered` or `ordered_scan` here, in case people confuses it with `ORDER BY` SQL semantic.